File: src/Faqt.GlobalConfigTests/TestUtils.fs
[<AutoOpen>]
module TestUtils

open Faqt
open Xunit


let assertExnMsg (msg: string) (f: unit -> 'a) =
    let ex = Assert.Throws<AssertionFailedException>(f >> ignore)

    Assert.Equal(
        ("\n\n" + msg.ReplaceLineEndings("\n").Trim() + "\n") :> obj, // Cast to obj to force full output
        ("\n\n" + ex.Message.ReplaceLineEndings("\n").Trim() + "\n")
    )

File: src/Faqt.CSharpTests/TestUtils.cs
using System;
using Faqt;
using Xunit;

internal class TestUtils
{
  internal static void AssertExnMsg(Action f, string msg)
  {
    var ex = Assert.Throws<AssertionFailedException>(f);
    Assert.Equal(
      msg.ReplaceLineEndings("\n").Trim(),
      ex.Message.ReplaceLineEndings("\n").Trim().Replace("\t", "    "));
  }
}
